### PR TITLE
Refactor snapshot agent to use new acl authmethod workflow

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Refactor Snapshot Agent

How I've tested this PR:
- Unit Tests
- ~Acceptance Tests~ Manually deployed it with ACLs enabled and ensured things came up without an error.

How I expect reviewers to test this PR:
- Code Review

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/client-snapshot-agent-deployment.yaml [hunk 1]
```diff
@@ -62,139 +62,178 @@ spec:
...
+          preStop:
+            exec:
+              command:
+              - "/bin/sh"
+              - "-ec"
+              - |
+                consul-k8s-control-plane consul-logout \
+                  {{- if .Values.global.adminPartitions.enabled }}
+                -partition={{ .Values.global.adminPartitions.name }} \
+                  {{- end }}
```
Comment (kschoche) [inline charts/consul/templates/client-snapshot-agent-deployment.yaml:159]: It is possible that this is a display issue on my side, but the text alignment seems to be off on these lines.

**Review by kschoche — APPROVED** [review]
Awesome work, I left a couple comments and one fix that needs to be addressed yet on consul-logout.
Most excellent work!

File: charts/consul/templates/client-snapshot-agent-deployment.yaml [hunk 2]
```diff
@@ -62,139 +62,175 @@ spec:
...
+        env:
+        - name: HOST_IP
+          valueFrom:
+            fieldRef:
+              fieldPath: status.hostIP
+          {{- if .Values.global.tls.enabled }}
+        - name: CONSUL_CACERT
+          value: /consul/tls/ca/tls.crt
+          {{- end }}
+        - name: CONSUL_HTTP_ADDR
```
Comment (ishustava) [inline charts/consul/templates/client-snapshot-agent-deployment.yaml:202]: need to fix indentation pls 🙏

File: charts/consul/templates/client-snapshot-agent-deployment.yaml [hunk 3]
```diff
@@ -62,139 +62,175 @@ spec:
...
+        {{- if (or .Values.global.acls.manageSystemACLs .Values.global.tls.enabled (and .Values.client.snapshotAgent.configSecret.secretName .Values.client.snapshotAgent.configSecret.secretKey) (and .Values.global.enterpriseLicense.secretName .Values.global.enterpriseLicense.secretKey .Values.global.enterpriseLicense.enableLicenseAutoload)) }}
+        {{- if .Values.global.acls.manageSystemACLs }}
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - "/bin/sh"
+              - "-ec"
+              - |
+                consul-k8s-control-plane consul-logout
```
Comment (ishustava) [inline charts/consul/templates/client-snapshot-agent-deployment.yaml:156]: this container is running with the consul image, and so this command is not available

**Comment by ishustava** [comment]
@thisisnotashwin 

>Acceptance Tests
there's no acceptance tests for the snapshot agent, so I think it needs to be tested manually

**Comment by thisisnotashwin** [comment]
> @thisisnotashwin
> 
> > Acceptance Tests
> 
> there's no acceptance tests for the snapshot agent, so I think it needs to be tested manually

ohno! i just assumed all tests going green was enough. sigh.

**Comment by thisisnotashwin** [comment]
> @thisisnotashwin
> 
> > Acceptance Tests
> 
> there's no acceptance tests for the snapshot agent, so I think it needs to be tested manually

DONE :)

**Review by ishustava — APPROVED** [review]

**Pull request merged**